pieman/bin/bscd.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/python3
+# Copyright (C) 2019 Evgeny Golyshev <eugulixes@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""BSC server. BSC stands for Build Status Codes.
+The server is very simple and supposed to handle one client at a time.
+"""
+
+import logging
+import os
+import socket
+import sys
+from argparse import ArgumentParser
+
+from daemon import DaemonContext, pidfile
+from redis.exceptions import RedisError
+
+from pieman import codes, util
+
+BUFSIZE = 4096
+
+DAEMON_NAME = 'bscd'
+
+LOGGER = logging.getLogger(DAEMON_NAME)
+
+
+def test_write_perm(dir_name):
+    """Tests if the current process has enough permissions to create files
+    in the specified directory.
+    """
+
+    dir_name = dir_name if dir_name else './'
+    if not os.access(dir_name, os.W_OK):
+        LOGGER.fatal('%s is not writable for the process. Make sure that the '
+                     'directory exists and the process has enough permissions '
+                     'to create files there.', dir_name)
+        sys.exit(1)
+
+
+class BSCServer:
```

Comment:
Not sure if Pieman repo is a suitable palce for this server

Reply:
The server is an integral part of Pieman, so there's no better place for it than Pieman=)